pkg/tests/base/render_test.go

```diff
@@ -86,24 +88,42 @@ func TestRenderUpstream(t *testing.T) {
 			template.TestingDisableKurlValues = true
 			defer func() { template.TestingDisableKurlValues = false }()
 
-			gotBase, _, err := base.RenderUpstream(&tt.Upstream, &tt.RenderOptions)
+			gotBase, gotHelmBases, err := base.RenderUpstream(&tt.Upstream, &tt.RenderOptions)
 			require.NoError(t, err)
 
-			if !assert.IsEqual(tt.WantBase, gotBase) {
-				t.Log(diffJSON(gotBase, tt.WantBase))
-			}
+			if len(tt.WantBase.Files) > 0 {
+				if !assert.IsEqual(tt.WantBase, gotBase) {
+					t.Log(diffJSON(gotBase, tt.WantBase))
+				}
 
-			if !assert.IsEqual(tt.WantBase.Files, gotBase.Files) {
-				for idx := range tt.WantBase.Files {
-					if len(gotBase.Files) > idx && gotBase.Files[idx].Path == tt.WantBase.Files[idx].Path {
-						t.Log("FILE", tt.WantBase.Files[idx].Path)
-						t.Log(diffString(string(gotBase.Files[idx].Content), string(tt.WantBase.Files[idx].Content)))
+				if !assert.IsEqual(tt.WantBase.Files, gotBase.Files) {
+					for idx := range tt.WantBase.Files {
+						if len(gotBase.Files) > idx && gotBase.Files[idx].Path == tt.WantBase.Files[idx].Path {
+							t.Log("FILE", tt.WantBase.Files[idx].Path)
+							t.Log(diffString(string(gotBase.Files[idx].Content), string(tt.WantBase.Files[idx].Content)))
+						}
+					}
+					t.FailNow()
+				}
+			}
+			if len(tt.WantHelmBase.Files) > 0 {
```

Comment:
This check shouldn't be needed.  Number of items in `gotHelmBases` should be checked instead